runtime-sdk/src/crypto/signature/ed25519.rs

```diff
@@ -1,26 +1,42 @@
 //! Ed25519 signatures.
-use ed25519_dalek::{self, ed25519::signature::Signature as _};
+use curve25519_dalek::edwards::CompressedEdwardsY;
 use serde::{Deserialize, Serialize};
 
-use oasis_core_runtime::common::crypto::{hash::Hash, signature::PublicKey as CorePublicKey};
+use oasis_core_runtime::common::crypto::signature::{
+    PublicKey as CorePublicKey, Signature as CoreSignature,
+};
 
 use crate::crypto::signature::{Error, Signature};
 
 /// An Ed25519 public key.
 #[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
-pub struct PublicKey(ed25519_dalek::PublicKey);
+pub struct PublicKey(CorePublicKey);
 
 impl PublicKey {
     /// Return a byte representation of this public key.
     pub fn as_bytes(&self) -> &[u8] {
-        self.0.as_bytes()
+        self.0.as_ref()
     }
 
     /// Construct a public key from a slice of bytes.
     pub fn from_bytes(bytes: &[u8]) -> Result<Self, Error> {
-        Ok(PublicKey(
-            ed25519_dalek::PublicKey::from_bytes(bytes).map_err(|_| Error::MalformedPublicKey)?,
-        ))
+        // CorePublicKey::from doesn't support error checking.
```

Comment:
Could implement `TryFrom` or `from_bytes` for the core public key and move this there.